frame/support/src/storage/mod.rs

```diff
@@ -472,9 +521,27 @@ mod test {
 
 			assert_eq!(MyStorage::iter().collect::<Vec<_>>(), vec![1, 2, 3, 4]);
 
+			// test removal
 			MyStorage::remove_all();
-
 			assert_eq!(MyStorage::iter().collect::<Vec<_>>(), vec![]);
+
+			// test migration
+			unhashed::put(&[&k[..], &vec![1][..]].concat(), &1u128);
```

Comment:
You inserting a `u128` and use `u32` in the translate function. You should actually store a `u32`.

And then call `iter()` to make sure that an empty vec is returned.

Than you should to the translation.

Reply:
oh yes I updated tests